src/open_inwoner/qmatic/client.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime
+
+from ape_pie.client import APIClient
+from pydantic import BaseModel
+from zgw_consumers.client import build_client
+from zgw_consumers.models import Service
+
+from open_inwoner.utils.api import JSONEncoderMixin
+
+from .exceptions import QmaticException
+from .models import QmaticConfig
+
+# API DATA DEFINITIONS
+
+
+class ServiceDict(BaseModel):
+    publicId: str
+    name: str
+    # could be float too in theory, documentation is not specific (it gives an int example)
+    duration: int
+    additionalCustomerDuration: int
+    custom: str | None
+
+
+class FullServiceDict(ServiceDict):
+    active: bool
+    publicEnabled: bool
+    created: int
+    updated: int
+
+
+class ServiceGroupDict(BaseModel):
+    services: list[ServiceDict]
+
+
+class BranchDict(BaseModel):
+    branchPublicId: str
+    branchName: str
+    serviceGroups: list[ServiceGroupDict]
+
+
+class BranchDetailDict(BaseModel):
+    name: str
+    publicId: str
+    phone: str | None
+    email: str | None
+    branchPrefix: str | None
+
+    addressLine1: str | None
+    addressLine2: str | None
+    addressZip: str | None
+    addressCity: str | None
+    addressState: str | None
+    addressCountry: str | None
+
+    latitude: float | None
+    longitude: float | None
+    timeZone: str
+    fullTimeZone: str
+    custom: str | None
+    created: int
+    updated: int
+
+
+class Appointment(JSONEncoderMixin, BaseModel):
+    services: list[ServiceDict]
+    title: str
+    start: datetime
+    end: datetime
+    created: int
+    updated: int
+    publicId: str
+    branch: BranchDetailDict
+    notes: str | None
+
+    class Config(JSONEncoderMixin.Config):
+        pass
+
+
+class NoServiceConfigured(RuntimeError):
+    pass
+
+
+# API CLIENT IMPLEMENTATIONS, per major version of the API
+
+
+def QmaticClient() -> "Client":
+    """
+    Create a Qmatic client instance from the database configuration.
+    """
+    config = QmaticConfig.get_solo()
+    assert isinstance(config, QmaticConfig)
+    if (service := config.service) is None:
+        raise NoServiceConfigured("No Qmatic service defined, aborting!")
+    assert isinstance(service, Service)
+    return build_client(service, client_factory=Client)
```

Comment:
if (service := config.service) is None:

I'm having doubts if this is a good pattern or possibly too clever walrus abuse. Usually you do the walrus so the assignment is there in the if-body, but this flips it around.

Reply:
I agree, I've been a bit too eager copy pasting :sweat_smile:. I'll switch it around such that the assignment variable is used inside the if statement.